### DISCUSSION TITLE
Dynamic AssistantId Setting

### DISCUSSION THREAD

**Discussion by drkfmorton** [boldare/openai-assistant#82] [Q&A]
Hi,

First off, thank you for such amazing work. As I am using the library, I am wondering if there is a way to dynamically set the assistant Id after it loads from the .env file?

**Comment by sebastianmusial** [comment]
Hello @drkfmorton , thank you for using the library and for your kind words!

Yes, you can dynamically set the assistant ID after loading it from the .env file. Here's an outline of how you can achieve this:

## Step 1: Import the AssistantModule
You can import the `AssistantModule` without any additional configuration. Once the assistant ID is available, you can pass it to subsequent methods.

## Step 2: Creating an Assistant
Here’s an example of how you can create an assistant instance:
```ts
import { AiService } from '@boldare/openai-assistant';

....

constructor(private readonly aiService: AiService) {}

create() {
  const assistant = this.aiService.provider.beta.assistants.create({ 
    ... // config
  });
 
 // Now you can use your assistant to the next actions
}

```

## Step 3: API Reference and payload details

Then can create the thread and pass the assistant ID when you want to send the message:
* the API [payload structure](https://assistant.ai.boldare.dev/api/docs#/Chat/ChatController_call)
* when invoking the `callStart` method: Make sure the `assistantId` is included in the payload. Please check the [method](https://github.com/boldare/openai-assistant/blob/main/apps/spa/src/app/modules/%2Bchat/shared/chat-gateway.service.ts#L25) and the [payload structure](https://github.com/boldare/openai-assistant/blob/main/libs/openai-assistant/src/lib/chat/chat.model.ts#L66).

This approach allows you to dynamically load the assistant ID and use it throughout your application. Let us know if you need further assistance! :)

**Comment by drkfmorton** [comment]
Thank you!!!!
